resources/guidelines/testing/store/quality-guidelines-plugins/README.md

```diff
@@ -0,0 +1,644 @@
+
+### Extension master data/license
+
+Please enter the valid license you set in your Shopware account. You have to identify this license in the manifest.xml as well.
+
+{% hint style="information" %}
+The chosen license can't be changed after adding your app to your account. If you want to change the license later, you must add a new app based on the app system with a new technical name and upload the extension again.
+{% endhint %}
+
+### Fallback language
+
+The installation is not always in English or German. So make sure that your app works in other languages as well.
+For example, if the customer has his installation in Spanish and your app is not yet available in this language, you should use the English translation as a fallback. Our test environment includes Netherland as the standard language.
+
+### Translations
+
+If your app is available in more than one language (e.g., English and German), these can be defined using the option "Translations into the following languages are available" (located in the “Description & images” section of your *Extension Manager*).
+
+### Valid app favicon for the Shopware Administration
+
+You must upload a valid favicon named plugin.png (png / 40 x 40 px) for the app. This favicon will make it easier to identify your app in the *Extension Manager* module in the backend. The favicon has to be stored under src/Resources/config/.
+
+### Error messages must be entered in the event log
+
+Error/informational messages can only be recorded in the event log of Shopware's log folder ( /var/log/ ). You have to develop your own log service. Never write app exceptions into the Shopware-Default-log or outside the Shopware system log folder. This ensures that the log file can never be accessed via the URL.
+
+{% hint style="danger" %}
+Avoid 400/500 errors at any time unless the 400 errors are related to an API call.
+{% endhint %}
+
+### Untrusted content should not be included
+
+See [untrusted content should not be included in SonarQube rules](https://rules.sonarsource.com/javascript/RSPEC-2611).
+
+###  Own media folder
+
+If it is possible for the customer to upload images, own media folders should be created, or existing ones should be used.
+
+### With "Install/Uninstall" the user must decide whether the data/tables are to be deleted or not
+
+When clicking on the "Install / Uninstall" option in the Extension Manager, the user must be presented with the options "completely delete" or "keep the app data, text snippets, and table adjustments".
+
+### It is not allowed to extend the Extension Manager
+
+The *Extension Manager* must not be extended or overwritten.
+
+### Extension manager
+
+The Debug Console controls the app's reinstallation, uninstallation, installation, and deletion. No 400 errors or exceptions are allowed to appear.
+If the app requires special PHP options, these must be queried during installation. If the query is negative, a growl message must appear in the backend.
```

Comment:
Does growl message mean as error message ?

Reply:
Yes, with growl message, we mean error message. Enclosed is a screenshot of an example of the shopware 5 growl messages.
![gorwl](https://user-images.githubusercontent.com/119946268/209789914-6f2075d4-d656-460f-97b4-00943e2ea530.png)